processor.go

```diff
@@ -398,14 +406,14 @@ func (p *processor) send(clientIP net.Addr, reqID uuid.UUID, tenant string, wr *
 }
 
 func (p *processor) fillRequestHeaders(
-	clientIP net.Addr, reqID uuid.UUID, tenant string, req *fh.Request) {
+	clientIP net.Addr, reqID uuid.UUID, tenantPrefix string, tenant string, req *fh.Request) {
 	req.Header.Set("Content-Encoding", "snappy")
 	req.Header.Set("Content-Type", "application/x-protobuf")
 	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
 	req.Header.Set("X-Cortex-Tenant-Client", clientIP.String())
 	req.Header.Set("X-Cortex-Tenant-ReqID", reqID.String())
-	if p.cfg.Tenant.Prefix != "" {
-		tenant = p.cfg.Tenant.Prefix + tenant
+	if tenantPrefix != "" {
+		tenant = tenantPrefix + tenant
```

Comment:
Why do we need to pass the prefix that deep? I think it's enough just to pass it to `dispatch()` and there prefix the tenants that are passed to `p.send()` with it.

Reply:
Hm, yeah I hadn't considered that. I could just override `p.cfg.Tenant.Prefix` when `PrefixPreferSource` is set. That'd save a lot of these changes.

I'll make those changes and will validate in my setup here.

Reply:
I've made these changes, and tested them in my setup and all looks good. And yeah, this _dramatically_ simplifies the change. Thanks for pushing on this one!

Reply:
OK, I believe I have addressed your concerns in https://github.com/blind-oracle/cortex-tenant/pull/62/commits/2e8c2fac7c6628e32f2f8a22defe4c584dea034b. Please confirm.